### PR TITLE
Config API-Calls and 405 Responses

### LINKED ISSUE #66: API: Return 405's instead of 404's when using wrong method
~~Appearently there seems to be a problem with sub-paths in the API. Calling /stream works on my end while /stream/restart results in an HTTP 404.~~

Maybe we should respond with an HTTP 405 when using GET on POST API calls and vice versa.

### LINKED ISSUE #67: API: Changing config
Consider an API-call to update content in the config.json as clients might have no write-permission on an external ressource.

### PR DESCRIPTION
**Pull request by AndreasWebdev**
This PR fixes #67 and fixes #66.

I've added two new API calls
* GET /config
* POST /config

And added missing API calls for not-supported methods with a simple 405 response

### CODE REVIEW

File: src/api/config.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+const authService = require('./auth');
+const upath = require('upath');
+const fs = require('fs-extra');
+const editJsonFile = require('edit-json-file');
+
+const getFullConfig = async (path, config) => {
```
Comment (torch2424) [inline src/api/config.js]: We need to re-require here ? 😮

File: src/api/config.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  // Return Config
+  return [200, config];
+};
+
+// Neat filter function for recursive object idenfitier
+function index(obj, i) {
+  return obj[i];
+}
+
+const getConfigByKey = async (path, config, key) => {
```
Comment (torch2424) [inline src/api/config.js]: We also would need to re-require the config here 😮

**Comment by AndreasWebdev** [comment]
You're right, thank you for the changes! Thankfully edit-json-file has a super-nice function for that anyways, which also means I was able to ditch the complicated "getByKey" method.

**Comment by torch2424** [comment]
Thanks for fixing that! And thanks again for all the help! 😄

**Pull request merged**